pagerduty/service.go

```diff
@@ -51,8 +51,8 @@ type AlertGroupingParameters struct {
 
 // AutoPauseNotificationsParameters defines how alerts on this service are automatically suspended for a period of time before triggering, when identified as likely being transient.
 type AutoPauseNotificationsParameters struct {
-	Enabled bool `json:enabled,omitempty`
-	Timeout int  `json:timeout,omitempty`
+	Enabled bool `json:"enabled"`
```

Comment:
Removing `omitempty` is required to avoid the following error when `enabled` is `false`:

```
---[ REQUEST ]---------------------------------------
PUT /services/P49BQU7 HTTP/1.1
Host: api.pagerduty.com
User-Agent: heimweh/go-pagerduty(terraform)
Content-Length: 369
Accept: application/vnd.pagerduty+json;version=2
Authorization: Token token=******
Content-Type: application/json
Accept-Encoding: gzip

{
 "service": {
  "acknowledgement_timeout": 1800,
  "alert_creation": "create_alerts_and_incidents",
  "alert_grouping": null,
  "auto_pause_notifications_parameters": {
   "timeout": 120
  },
  "auto_resolve_timeout": 1800,
  "description": "foo",
  "escalation_policy": {
   "id": "PW95IJV",
   "type": "escalation_policy_reference"
  },
  "incident_urgency_rule": {
   "type": "constant",
   "urgency": "high"
  },
  "name": "tf-7vwjy"
 }
}

-----------------------------------------------------
2022/06/10 14:22:53 [DEBUG] PagerDuty API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 400 Bad Request
Content-Length: 116
Access-Control-Allow-Headers: Authorization, Content-Type, AuthorizationOauth, X-EARLY-ACCESS
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers:
Access-Control-Max-Age: 1728000
Cache-Control: no-cache
Content-Type: application/json
Date: Fri, 10 Jun 2022 12:22:53 GMT
Referrer-Policy: strict-origin-when-cross-origin
Server: nginx
X-Request-Id: c1841333efa37b0fd7acd858b7648312

{
 "error": {
  "message": "Invalid Input Provided",
  "code": 2001,
  "errors": [
   "Enabled must be a TrueClass or a FalseClass."
  ]
 }
}
-----------------------------------------------------
```